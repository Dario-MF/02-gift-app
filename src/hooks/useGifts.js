import { useEffect, useState } from 'react'
import { getGifts } from '../helpers/getGifts'

const useGifts = (category) => {
    const [state, setState] = useState({
        data: [],
        loading: true
    })

    useEffect(() => {
        getGifts(category)
            .then(gifs => {

                setTimeout(() => {
                    setState({
                        data: gifs,
                        loading: false
                    })
                }, 3000);

            })

    }, [category])

    return state

}

export default useGifts
