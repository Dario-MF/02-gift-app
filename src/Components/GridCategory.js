import React from 'react'
import useGifts from '../hooks/useGifts'
import GridItem from './GridItem'
import PropTypes from 'prop-types'


const GridCategory = ({ category }) => {

    const { data: gifts, loading } = useGifts(category)

    return (
        <div>
            <h2>{category}</h2>
            {loading && <p>Cargando...</p>}
            <div className={'grid-cards'}>
                {
                    gifts.map(gift => (
                        <GridItem
                            key={gift.id}
                            {...gift}
                        />
                    ))
                }
            </div>
        </div>
    )
}

GridCategory.propTypes = {
    category: PropTypes.string.isRequired
}

export default GridCategory
