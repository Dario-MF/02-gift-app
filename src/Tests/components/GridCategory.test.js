import { shallow } from 'enzyme'
import React from 'react'
import '@testing-library/jest-dom'
import GridCategory from '../../Components/GridCategory'
import useGifts from '../../hooks/useGifts'
jest.mock('../../hooks/useGifts')

describe('Test for component <GridCategory />', () => {

    const category = 'One piece'
   
    test('snapshot for <GridCategory />', () => {
        useGifts.mockReturnValue({
                data:[],
                loading: true
            }
        )
        const wrapper = shallow(<GridCategory category={ category }/>)
        expect(wrapper).toMatchSnapshot()
    })

    test('evaluar que useGifts devuelve la data correctamente', () => {
        const gifs = [{
            id: '1234',
            title: 'titulo gift',
            url: 'https://image/gif.jpg'
        }]
        
        useGifts.mockReturnValue({
                data:gifs,
                loading: false
            }
        )
        const wrapper = shallow(<GridCategory category={ category }/>)

        expect(wrapper).toMatchSnapshot()
        expect(wrapper.find('p').exists()).toBe(false)
        expect(wrapper.find('.grid-cards').length).toBe(gifs.length)
    })
    
    
})
