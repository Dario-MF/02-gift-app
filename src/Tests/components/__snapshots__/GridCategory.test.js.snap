// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for component <GridCategory /> evaluar que useGifts devuelve la data correctamente 1`] = `
<div>
  <h2>
    One piece
  </h2>
  <div
    className="grid-cards"
  >
    <GridItem
      id="1234"
      key="1234"
      title="titulo gift"
      url="https://image/gif.jpg"
    />
  </div>
</div>
`;

exports[`Test for component <GridCategory /> snapshot for <GridCategory /> 1`] = `
<div>
  <h2>
    One piece
  </h2>
  <p>
    Cargando...
  </p>
  <div
    className="grid-cards"
  />
</div>
`;
